artemis-kryo/src/main/java/com/artemis/io/KryoEntitySerializer.java

```diff
@@ -230,10 +230,9 @@ protected void clearSerializerCache() {
 		serializers.clear();
 	}
 
-	// TODO do we need to expose this for subclassing?
-	public static class EditFieldSerializer extends FieldSerializer {
+	public static class EditFieldSerializer<T extends Component> extends FieldSerializer<T> {
```

Comment:
i like this solution; this way, we have a venue for dealing with versioning of component types  + optimized serializers (sometime in the future, we could auto-gen these)